### PR TITLE
Update keen tracking

### PR DESCRIPTION
**Pull request by dustinlarimer**
## What does this PR do?

This PR updates the Keen IO integration to mitigate the potential for namespace collisions when using other versions of Keen SDKs.
- Update keen-js to v3.1.0 (latest possible without breaking changes)
- Create a new `KeenSegment` namespace for analytics.js-installed SDK and return existing `Keen` object if overwritten

### CODE REVIEW

**Comment by dustinlarimer** [comment]
@ndhoule @sperand-io created a patch in this PR that should help customers work around current namespace issues. Could you offer any guidance on how to fix this build error?

**Comment by dustinlarimer** [comment]
cc @hankim813

File: lib/index.js [hunk 1]
```diff
@@ -10,51 +11,48 @@ var clone = require('clone');
...
  */
 
-var Keen = module.exports = integration('Keen IO')
+var KeenSegment = module.exports = integration('Keen IO')
   .global('Keen')
+  .global('KeenSegment')
   .option('ipAddon', false)
   .option('projectId', '')
-  .option('readKey', '')
+  .option('readKey', false)
```
Comment (hankim813) [inline lib/index.js:19]: We can't make the default boolean here because our metadata service sets this option's type as a string. If we make it a boolean type, the settings will change into a checkbox and since we need to still grab the readkey string, you'll have to leave the default value as an empty string.

File: lib/index.js [hunk 2]
```diff
@@ -10,51 +11,48 @@ var clone = require('clone');
...
+  this.load({ lib: lib }, function() {
+    window.KeenSegment = window.Keen;
+    self.client = new window.KeenSegment({
+      projectId: options.projectId,
+      readKey: options.readKey,
+      writeKey: options.writeKey
+    });
+    self.ready();
+    if (previousKeen) {
+      window.Keen = previousKeen;
```
Comment (hankim813) [inline lib/index.js:53]: Could you explain this part? I'm slightly confused how this works... sorry!

File: lib/index.js [hunk 3]
```diff
@@ -130,10 +128,10 @@ Keen.prototype.identify = function(identify) {
  * @param {Track} track
  */
 
-Keen.prototype.track = function(track) {
+KeenSegment.prototype.track = function(track) {
   var props = track.properties();
   this.addons(props, track);
-  this.client.addEvent(track.event(), props);
+  this.client.recordEvent(track.event(), props);
```
Comment (hankim813) [inline lib/index.js:134]: Any notable changes in regards to functionalities between these two functions? Or is it just an syntactical update?

File: test/index.test.js [hunk 4]
```diff
@@ -12,12 +13,11 @@ describe('Keen IO', function() {
     projectId: '53e40374e861700e79000002',
     writeKey: 'f474c11bcf5813fabc933d0a1f80ff2e1ac9ff1c9338dcf2707bbb8cf68971ae524b4cac8db5da549aa3383d499ed4a809e2b6294d1d093818d699495a4171dab3e6ad86e2c31d06627f27aadb8433f43c8f27f3e0354e6ea4c12f9da57ca9d8420b18c5e4f719286a0fae08914d9c44'
   };
-  var readKey = 'e5cdee9b7395b315bd8cc635f3b04fc07561d4e42889fe1b6ac719a9a4b45732c746666d83ce8644c8b0f06b867166654f900b67b250adbc225befac4b3a2562729c2ebebfbf19b1d13f631c2ed0c9f8de0be7897eded88102abe4366c7906011dd480631ed9ba60cdef84f908abc852';
```
Comment (hankim813) [inline test/index.test.js:15]: Why was this removed?

File: test/index.test.js [hunk 5]
```diff
@@ -30,11 +30,11 @@ describe('Keen IO', function() {
...
 
   it('should have the right settings', function() {
-    analytics.compare(Keen, integration('Keen IO')
-      .global('Keen')
+    analytics.compare(KeenLib, integration('Keen IO')
+      .global('KeenSegment')
       .option('ipAddon', false)
       .option('projectId', '')
-      .option('readKey', '')
+      .option('readKey', false)
```
Comment (hankim813) [inline test/index.test.js:37]: Should change back to a string as per noted above

File: test/index.test.js [hunk 6]
```diff
@@ -49,46 +49,26 @@ describe('Keen IO', function() {
...
-          readKey: ''
+        analytics.ready(function() {
+          analytics.equal(keen.client.projectId(), options.projectId);
+          analytics.equal(keen.client.writeKey(), options.writeKey);
         });
       });
     });
   });
 
-  describe('loading', function() {
```
Comment (hankim813) [inline test/index.test.js:71]: Can we also add a test somewhere that checks that whole behavior regarding `previousKeen`?

**Comment by hankim813** [comment]
As far as the `syntax parse error` this is due to a wildcard dependency in your `component.json` file.

Can you change this line: https://github.com/keen/analytics.js-integration-keen-io/blob/84427c056922350d7ebb094c794175136333309e/component.json#L10

To this? https://github.com/segmentio/clone/tree/0.1.1

So basically you want that line to be `"segmentio/clone": "0.1.1"`

Then locally you want to run `make distclean` to remove the `build.js` and the `components/` directory. Then you should be able to run `make test` or `make test-browser`. 

Let me know if you still have build issues! 

Thanks again for the PR!

**Comment by dustinlarimer** [comment]
@hankim813 sorry for the mess there.. I was working from an earlier branch where I attempted to install keen-tracking.js, but had decided against proposing it. I'll open a fresh PR now~
